### PR TITLE
Enable multiple package managers in environment window

### PR DESCRIPTION
**Pull request by huguesv**
Small incremental improvement that lets you use either pip or conda in the environment window (when conda is detected).

I've avoided changing the interface, which wasn't technically necessary to support this particular enhancement. I think this will have to be revisited (I don't like it much).

Other enhancements may require access to the alternate package manager, which may mean we need it stored on the factory interface. See my latest notes #3350 for various things that come up when pip is not the default package manager.

### CODE REVIEW

File: Python/Product/Analysis/Interpreter/Ast/AstPythonInterpreterFactory.cs [hunk 1]
```diff
@@ -70,6 +70,7 @@ InterpreterFactoryCreationOptions options
                     if (pm != null) {
                         pm.SetInterpreterFactory(this);
                         pm.InstalledFilesChanged += PackageManager_InstalledFilesChanged;
+                        // TODO: do we need to worry about the non-primary package managers?
```
Comment (zooba) [inline Python/Product/Analysis/Interpreter/Ast/AstPythonInterpreterFactory.cs]: No. The point of this one is to be the primary package manager (mainly for detecting "do we need to rescan for intellisense")

File: Python/Product/PythonTools/PythonTools/InterpreterList/InterpreterListToolWindow.cs [hunk 2]
```diff
@@ -244,6 +240,16 @@ sealed class InterpreterListToolWindow : ToolWindowPane {
             }
         }
 
+        private void AddPepExtension(EnvironmentView view, IPythonInterpreterFactory factory, IPackageManager packageManager) {
```
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/InterpreterList/InterpreterListToolWindow.cs]: `AddPipExtension`

File: Python/Product/VSInterpreters/PackageManager/PackageManagers.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.PythonTools.Interpreter {
+    public static class PackageManagers {
```
Comment (zooba) [inline Python/Product/VSInterpreters/PackageManager/PackageManagers.cs]: We should have an extension methods class somewhere, right? Can we have `GetAllPackageManagers` that returns all of them?

**Comment by huguesv** [comment]
Thanks, updated.

**Comment by zooba** [comment]
The changes are good, but let's chat about refactoring package manager management more before merging it.

**Comment by huguesv** [comment]
The change was integrated in #3500

**Pull request closed without merging**